docs/algorithms/cost_generic_resub.rst

```diff
@@ -0,0 +1,25 @@
+.. _cost_generic_resub:
+
+Cost-generic resubstitution algorithm
+-------------------------
+
+**Header:** ``mockturtle/algorithms/experimental/cost_generic_resub.hpp``
+
+This header file defines a resubstitution algorithm to optimize a network with a customized cost function. 
+
+.. code-block:: c++
+
+   /* derive some XAG */
+   xag_network xag = ...;
+
+   cost_generic_resub_params ps;
+   cost_generic_resub_stats st;
+
+   cost_generic_resub( xag, size_cost_function<xag_network>(), ps, &st );
```

Comment:
Can you also teach the users how to define their own cost function / point to the documentation for this?